coderd/workspaces.go

```diff
@@ -1029,6 +1029,55 @@ func (api *API) watchWorkspace(rw http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// @Summary Transfer workspace owner by ID
+// @ID transfer-workspace-owner-by-id
+// @Security CoderSessionToken
+// @Accept json
+// @Tags Workspaces
+// @Param workspace path string true "Workspace ID" format(uuid)
+// @Param request body codersdk.TransferWorkspaceOwnerRequest
+// @Success 204
+// @Router /workspaces/{workspace}/transfer [put]
+func (api *API) putTransferWorkspace(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx               = r.Context()
+		workspace         = httpmw.WorkspaceParam(r)
+		auditor           = api.Auditor.Load()
+		aReq, commitAudit = audit.InitRequest[database.Workspace](rw, &audit.RequestParams{
+			Audit:   *auditor,
+			Log:     api.Logger,
+			Request: r,
+			Action:  database.AuditActionWrite,
+		})
+	)
+
+	defer commitAudit()
+	aReq.Old = workspace
+
+	var req codersdk.TransferWorkspaceOwnerRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	now := database.Now()
+	if err := api.Database.UpdateWorkspaceOwnerByID(ctx, database.UpdateWorkspaceOwnerByIDParams{
```

Comment:
Is it possible to have multiple workspaces with the same name?

Reply:
Currently it's using the workspace ID, not name, for the transfer operation. I'd argue this isn't a great user experience, since an end user probably doesn't easily know the ID of their workspace. But since it's a UUID I'd feel confident there wouldn't be a duplication issue there. As for the name itself there could be a duplication, but because the display name isn't used as a unique key I don't see it as a problem in this particular case. It could be confusing, however, but not a technical issue.

Reply:
> As for the name itself there could be a duplication, but because the display name isn't used as a unique key I don't see it as a problem in this particular case. 

What about URL navigation where the workspace name is used as SLUG?